### PR TITLE
Attachments review

### PR DESCRIPTION
**Pull request by miggol**
This brings back missing Documents functionality on the review side:

* Documents list in the review sidebars
* Compare Attachments
* Secretary can edit Attachments
* New attachments get a little sun icon in the review sidebar. This is to distinguish them from attachments that were simply completely unchanged, which now no longer get a "compare" button.

Also, legacy attachments are appended to the bottom of the attachments page while we transition to the new system.

### CODE REVIEW

**Review by EdoStorm96 — DISMISSED** [review]
Impressive work overall! This is a good start, but I still have some issues. Mostly discussed in separate comments. Some general comments:
- I find the AttachmentsList (and all the parts that are needed for generating it) a bit overwhelming ... and feel like some of the functionality of these classes can be had, without so many new classes. Like can't the slots just be used directly instead of the AttachmentItem's? Maybe with some new methods? If you can convince me of this design though, we can keep it. As it does work ofc :) But I feel like it can be simpler (just don't quite know how ...) See my other comments for more explanation.
- A minor issue: The ReviewSidebarMixin has not been added to all views which feature the review sidebar (eg. DecisionUpdateView & ReviewCloseView). Also, it does not accommodate views which don't have a review as their object.

Otherwise nothing too crazy in my comments, just little things. Message me if you need further clarification :)

File: proposals/utils/stepper.py [hunk 1]
```diff
@@ -147,6 +147,17 @@ def check_all(self, next_checkers):
         # Recurse until next_checkers is empty
         return self.check_all(next_checkers)
 
+    def add_slot(self, slot):
```
[the remark this answers is not in the document]
Comment (miggol) [reply proposals/utils/stepper.py:150]: Lol what. How did this happen

File: reviews/templates/reviews/documents_list.html [hunk 2]
```diff
@@ -6,33 +6,37 @@
...
+                        {% endif %}
+                        {{ item.name }}
+                    </a>
+                    {% if item.comparable %}
+                        {% simple_compare_link item.object item.field %}
+                    {% endif %}
+                </li>
+            {% endfor %}
+        </ul>
+        {% if container.dmp_edit_link %}
```
[the remark this answers is not in the document]
Comment (miggol) [reply reviews/templates/reviews/documents_list.html:28]: Done!

File: reviews/templatetags/documents_list.py [hunk 3]
```diff
@@ -156,33 +156,42 @@ def __init__(self, header, **kwargs):
...
     def get_link_url(self):
         if self.link_url:
             return self.link_url
-
         return self.field.url
 
 
+@register.inclusion_tag("reviews/documents_list.html")
+def attachments_list(review, user):
+    stepper = Stepper(review.proposal)
```
[the remark this answers is not in the document]
Comment (miggol) [reply reviews/templatetags/documents_list.py:187]: Done!

File: reviews/views.py [hunk 4]
```diff
@@ -36,6 +37,7 @@
     assign_reviewers,
 )
 from .utils.review_actions import ReviewActions
+from attachments.utils import AttachmentsList
```
[the remark this answers is not in the document]
Comment (miggol) [reply reviews/views.py:40]: Fixed

File: locale/en/LC_MESSAGES/django.po [hunk 5]
```diff
@@ -1841,30 +2043,62 @@ msgstr ""
 "Documents can also be compared with a combined view, with markings. To do "
 "so, click 'View combined'."
 
+#: proposals/templates/proposals/compare_attachments.html:67
 #: proposals/templates/proposals/compare_documents.html:67
 msgid "Je vergelijkt"
 msgstr "Comparing:"
```
[the remark this answers is not in the document]
Comment (miggol) [reply locale/en/LC_MESSAGES/django.po:2049]: Fixed

File: attachments/utils.py [hunk 6]
```diff
@@ -124,6 +125,150 @@ def get_edit_url(
         )
 
 
+class AttachmentItem(DocItem):
+
```
[the remark this answers is not in the document]
Comment (miggol) [reply attachments/utils.py:129]: > I just find it a bit confusing. Also, where does self.slot get defined in this class?>

```python
class DocItem:
    def __init__(self, header, **kwargs):
        ...
        self.__dict__.update(kwargs)
```

This last part defines anything else given as a `kwarg` as a class variable. This is a common pattern for "container" classes whose primary function is to just contain data in a structured way.

**Pull request merged**